src/app/collective-rewards/allocations/AllocationAmount.tsx

```diff
@@ -0,0 +1,74 @@
+'use client'
+
+import { formatBalanceToHuman } from '@/app/user/Balances/balanceUtils'
+import { Button, ButtonProps } from '@/components/Button'
+import { Input } from '@/components/Input'
+import { Link } from '@/components/Link'
+import { cn } from '@/lib/utils'
+import { useState } from 'react'
+import { StakeHint } from './StakeHint'
+
+const PercentageButton = ({ children, variant, ...rest }: ButtonProps) => (
+  <Button
+    className={cn('px-2 py-1', variant === 'primary' && 'border-[1px] border-primary')}
+    variant={variant}
+    {...rest}
+  >
+    {children}
+  </Button>
+)
+
+type AllocationAmountProps = {
+  balance: bigint
+  errorMessage?: string
+  onPercentageSet?: (percentage: number) => void
+}
+
+export const AllocationAmount = ({ balance, errorMessage, onPercentageSet }: AllocationAmountProps) => {
+  // TODO: hint is shown only when the allocated amount exceeds the balance
+  // const hint = <StakeHint />
```

Comment:
?

Reply:
I've left them just as placeholder, so when it's time to implement the logic, we already know what to change